miscellaneous/roboconf-integration-tests-dm-with-agents-in-memory/src/test/java/net/roboconf/integration/tests/dm/with/agents/in/memory/ExecuteScriptResourcesTest.java

```diff
@@ -0,0 +1,149 @@
+import net.roboconf.core.model.helpers.InstanceHelpers;
+import net.roboconf.core.utils.Utils;
+import net.roboconf.dm.internal.utils.ConfigurationUtils;
+import net.roboconf.dm.management.ManagedApplication;
+import net.roboconf.integration.tests.commons.AbstractIntegrationTest;
+import net.roboconf.integration.tests.commons.ItConfigurationBean;
+import net.roboconf.integration.tests.commons.internal.parameterized.IMessagingConfiguration;
+import net.roboconf.integration.tests.commons.internal.runners.RoboconfPaxRunner;
+import net.roboconf.integration.tests.dm.with.agents.in.memory.internal.MyHandler;
+import net.roboconf.integration.tests.dm.with.agents.in.memory.internal.MyTargetResolver;
+import net.roboconf.integration.tests.dm.with.agents.in.memory.probes.DmWithAgentInMemoryTest;
+
+/**
+ * Test a script execution by an agent.
+ * <p>
+ * Deploy and start all instances, etc.
+ * On two agents.
+ * </p>
+ *
+ * @author Amadou Diarra - UGA
+ */
+@RunWith( RoboconfPaxRunner.class )
+@ExamReactorStrategy( PerMethod.class )
+public class ExecuteScriptResourcesTest extends DmWithAgentInMemoryTest {
+
+	private static final String APP_LOCATION = "my.app.location";
+
+	@ProbeBuilder
+	public TestProbeBuilder probeConfiguration( TestProbeBuilder probe ) {
+
+		// We need to specify the classes we need
+		// and that come from external modules.
+		probe.addTest( DmWithAgentInMemoryTest.class );
+		probe.addTest( TestUtils.class );
+
+		probe.addTest( AbstractIntegrationTest.class );
+		probe.addTest( IMessagingConfiguration.class );
+		probe.addTest( ItConfigurationBean.class );
+
+		probe.addTest( MyHandler.class );
+		probe.addTest( MyTargetResolver.class );
+
+		return probe;
+	}
+
+
+	@Override
+	@Configuration
+	public Option[] config() throws Exception {
+
+		File resourcesDirectory = TestUtils.findApplicationDirectory( "simple" );
+		String appLocation = resourcesDirectory.getAbsolutePath();
+		return OptionUtils.combine(
+				super.config(),
+				systemProperty( APP_LOCATION ).value( appLocation ));
+	}
+
+
+	@Test
+	public void run() throws Exception {
+
+		// Update the manager
+		configureManagerForInMemoryUsage();
+
+		// Load the application
+		String appLocation = System.getProperty( APP_LOCATION );
+		ApplicationTemplate tpl = this.manager.applicationTemplateMngr().loadApplicationTemplate( new File( appLocation ));
+		ManagedApplication ma = this.manager.applicationMngr().createApplication( "test", null, tpl );
+		Assert.assertNotNull( ma );
+		Assert.assertEquals( 1, this.manager.applicationMngr().getManagedApplications().size());
+
+		// Create script files
+		String targetId = this.manager.targetsMngr().findTargetId(ma.getApplication(), "/MySQL VM");
+		File dir = new File( this.manager.configurationMngr().getWorkingDirectory(), ConfigurationUtils.TARGETS + "/" + targetId );
+		Utils.writeStringInto( "#!/bin/bash\necho toto > toto.txt", new File( dir, "totoscript-all.sh"));
```

Comment:
"toto.txt" is created in /tmp/roboconf_agent/VM_NAME/ once th script is executed.